server/services/participants.js

```diff
@@ -13,15 +13,27 @@ const setParticipantStatus = async (
   data, // JSONB on the status row
 ) => dbClient.db.withTransaction(async (tx) => {
   if (status !== 'rejected') {
-    const item = await tx[collections.PARTICIPANTS_STATUS].find({
+    const items = await tx[collections.PARTICIPANTS_STATUS].find({
       participant_id: participantId,
       status: 'hired',
       current: true,
     });
 
-    if (item.length > 0) return { status: 'already_hired' };
+    if (items.length > 0) return { status: 'already_hired' };
   }
 
+  const item = await tx[collections.PARTICIPANTS_STATUS].findOne({
+    participant_id: participantId,
+    employer_id: employerId,
+    current: true,
+  });
+
+  if ((status === 'prospecting' && item !== null && item.status !== 'open')
+    || (status === 'interviewing' && item?.status !== 'prospecting')
+    || (status === 'offer_made' && item?.status !== 'interviewing')
+    || (status === 'hired' && item?.status !== 'offer_made')
+  ) return { status: 'invalid_status_transition' };
+
```

Comment:
I think this is difficult to tell exactly what the criteria are for each status.

So here's how I'm reading this (status refers to the target/intended status for a participant):
- Rejecting a participant is allowed even if the participant has been hired elsewhere
- If engaging (prospecting), participant must be coming from open or null status
- If interviewing, particpant must be coming from prospecting status
- If offer made, must be coming from interviewing
- If hiring, must be coming from offer made